.github/workflows/test.yml

```diff
@@ -5,6 +5,10 @@ on: [push, pull_request]
 
 permissions: read-all
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Comment:
`github.ref` for pull requests [has the form](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context) of `refs/pull/<pr_number>/merge`, i.e. it already includes the pull request number.

Reply:
changes applied